tests/unit/test_download.py

```diff
@@ -157,6 +158,98 @@ def test_unpack_http_url_bad_downloaded_checksum(mock_unpack_file):
         rmtree(download_dir)
 
 
+def test_download_http_url_with_directory_traversal(tmpdir):
+    """
+    Test downloading a file when the content-disposition header
+    contains a filename with a ".." path part.
+    It should download file to the tmp_subdir.
+    """
+    mock_url = 'http://www.example.com/whatever.tgz'
+    contents = b'downloaded'
+    link = Link(mock_url)
+
+    session = Mock()
+    resp = MockResponse(contents)
+    resp.url = mock_url
+    resp.headers = {
+        # content-type set random avoid mimetypes.guess_extension guess ext
+        'content-type': 'random',
+        'content-disposition': 'attachment;filename="../out_dir_file"'
+    }
+    session.get.return_value = resp
+
+    tmp_subdir = mkdtemp(dir=tmpdir, prefix="sub-dir-")
+    file_path, content_type = _download_http_url(
+        link,
+        session,
+        tmp_subdir,
+        hashes=None,
+        progress_bar="on"
+    )
+    assert set(os.listdir(tmp_subdir)) == {
+        'out_dir_file'
+    }
+
+
+@pytest.mark.skipif("sys.platform != 'win32'")
+@pytest.mark.parametrize("filename, expected", [
+    ('dir/file', 'file'),
+    ('../file', 'file'),
+    ('../../file', 'file'),
+    ('../', ''),
+    ('../..', '..'),
+    ('/', ''),
+    ('dir\\file', 'file'),
+    ('..\\file', 'file'),
+    ('..\\..\\file', 'file'),
+    ('..\\', ''),
+    ('..\\..', '..'),
+    ('\\', ''),
+])
+def test_sanitize_content_filename_win32(filename, expected):
```

Comment:
My review comment before said one test should be for the test cases where the two test expectations are the same, and one for where they're different -- rather than one for Windows and non-Windows. With the current approach, it's not easy to know / check if the same test cases are being covered in both. This means that the first test will have one test expectation per parametrized test input, and the other will have two test expectations per test input Windows and non-Windows (like you had in the previous iteration of your PR).